iOS/MobileStarter/react-app/src/HubScreen.tsx

```diff
@@ -239,247 +114,109 @@ export function HubScreen(props: HubScreenProps) {
-      // First, check to see if we have a local copy of the iModel.
-      // NOTE: This does not check to see if the iModel has been updated since it was last
-      // downloaded.
-      const localBriefcases = await NativeApp.getCachedBriefcases(iModel.id);
-      if (!isMountedRef.current) return;
-      if (localBriefcases.length === 0) {
-        // We don't have a local copy of the iModel, so download it and then open it.
-        downloadIModel(iModel);
-      } else {
-        // We do have a local copy of the iModel, so open that local copy.
-        openIModel(localBriefcases[0]);
-      }
-    } catch (error) {
-      // There was a problem loading the iModel. Show the error, and give the user the option
-      // to sign out.
-      presentError("LoadErrorFormat", error);
-      setButtonTitles([signOutLabel]);
-    }
-  }, [iModels, downloadIModel, openIModel, isMountedRef, signOutLabel]);
-
-  let filteredTitles = buttonTitles;
-  if (filter && hubStep === HubStep.SelectProject) {
-    filteredTitles = buttonTitles.filter((value) => value.toLocaleLowerCase().indexOf(filter) !== -1);
-  }
-  // Convert the button titles into buttons.
-  const buttons = filteredTitles.map((title: string, index: number) => {
-    const getTitle = (title: string, briefcase: LocalBriefcaseProps | undefined) => {
-      if (!briefcase) return title;
-      return title + " (" + fileSizeString(briefcase.fileSize) + ")";
-    }
-    if (hubStep === HubStep.SelectIModel || hubStep === HubStep.DownloadIModel) {
-      const briefcase = iModels[index].briefcase;
-      let deleteButton;
-      if (briefcase) {
-        deleteButton = (
-          <div className="delete-button" onClick={async (e) => {
-            e.stopPropagation();
-            try {
-              await MobileCore.deleteCachedBriefcase(briefcase);
-              if (!isMountedRef.current) return;
-              if (project) {
-                selectProject(project);
-              }
-            } catch (error) {
-              // There was a problem deleting the cached briefcase. Show the error.
-              presentError("DeleteErrorFormat", error, "HubScreen");
-            }
-          }}>
-            <IconImage iconSpec="icon-delete" />
-          </div>
-        );
-      }
-      return (
-        <div className="imodel-row" key={index}>
-          <Button
-            onClick={() => handleSelectIModel(index)}
-            title={getTitle(title, briefcase)}
-          >
-            {deleteButton}
-          </Button>
-        </div>);
-    } else {
-      return <Button
-        key={index}
-        onClick={async () => {
-          if (title === signOutLabel) {
-            if (IModelApp.authorizationClient instanceof ITMAuthorizationClient) {
+  }, [isMountedRef, project, initialized]);
+
+  let moreButton: React.ReactNode;
+  let stepContent: React.ReactNode;
+
+  switch (hubStep) {
+    case HubStep.SelectProject:
+      stepContent = <ProjectPicker signedIn={initialized}
```

Comment:
I feel that there should be a HubStep.SigningIn (or some such) that shows a spinner and a message. It would be impossible to show the ProjectPicker or the IModelPicker before the sign in has succeeded. That simplifies those components, and also shows a useful spinner while sign in is occurring.

Reply:
Done, added a new SignIn component (in its own file)